UM/Settings/ContainerStack.py

```diff
@@ -605,6 +634,7 @@ def getSettingDefinition(self, key: str) -> Optional[SettingDefinition]:
             return None
 
     @UM.FlameProfiler.profile
+    #TODO: Find a way around keyword-arguments being passed as a dict, or live with not cacheing this particular func.
```

Comment:
Now the question is, how often is this method called in normal use conditions ?

Reply:
There's 100+ matches on 'findContainer' and they bascially never call without arguments. Each of them would have to be converted to using criteria instead of the kwargs as much as possible, and that would have to be a frozendict or somesuch (which is a separate library, as opposed to frozenset), which would also break the API, unless we make it a separate method and don't use this one. 🫠 

Seeing as the rest of the caching already does its job, I don't think its worth it 😅